### DISCUSSION TITLE
update then disconnect relationship

### DISCUSSION THREAD

**Discussion by laurent512** [prisma/prisma#13902] [Q&A]
Hello, 

I have 2 model : Wallet and TransactionKey. 
Each wallet can be associated to maximum one transactionKey (allowing temporary access to the wallet).
Once the transactionKey is consumed, it should be deleted.
 
Is there a way to update the wallet (e.g. update its balance), by using its associated transactionKey, and then disconnecting (or deleting) the associated transactionKey?

To make it more complex, the should work by having access to the TransactionKey id, not walletId.

Thank you !

**Comment by austincrim** [comment]
Hey @laurent512 👋,

Have you looked at [interactive transactions](https://www.prisma.io/docs/guides/performance-and-optimization/prisma-client-transactions-guide#interactive-transactions-in-preview)?

You could open up a transaction and then perform the update and the disconnect in sequence.

```ts
await prisma.$transaction(async (prisma) => {
  await prisma.wallet.update(...)
  await prisma.transactionKey.delete(...)
})
```

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
